pkg/prober/prober_unix.go

```diff
@@ -0,0 +1,24 @@
+//go:build !windows
+// +build !windows
+
+package prober
+
+import (
+	"crypto/x509"
+
+	"github.com/sirupsen/logrus"
+)
+
+// GetSystemCertPool returns a x509.CertPool that contains the
+// root CA certificates if they are present at runtime
+func GetSystemCertPool(probeName string) (*x509.CertPool, error) {
+	caCertPool, err := x509.SystemCertPool()
+	if err != nil {
+		caCertPool = x509.NewCertPool()
+		logrus.Errorf("[GetSystemCertPoolUnix] error loading system cert pool for probe (%s): %v", probeName, err)
+	}
+	if caCertPool == nil {
+		return nil, fmt.Errorf("[GetSystemCertPoolWindows] x509 returned a nil certpool for probe (%s)", probeName)
+	}
```

Comment:
I think this can be removed.

Reply:
I added the nil check inside each of the functions for linux/windows as it allows the function to fail faster for Windows and avoid a situation where a nil pointer causes a failure before we could check it in the main prober.go. Waiting to check until after the caCertPool is returned could result in a nil pointer being returned when we try to append the certs from the syscall to the caCertPool.